semgrep-core/src/core/ast/AST_generic.ml

```diff
@@ -352,6 +351,15 @@ and qualifier =
   (* Ruby/Lua *)
   | QExpr of expr * tok
 
+(* This is not relevant for matching purposes.
+   `unique_name` is a record of all the information necessary to
+   disambiguate two `ResolvedName`s from each other.
+   In particular, two `ResolvedName`s may indicate two overloaded functions,
+   which are named the same. Then, we need to use the function signature information
+   to disambiguate them from each other.
+*)
+and unique_name = { dotted : dotted_ident; id : id_info_id option }
```

Comment:
You've almost certainly explained this somewhere, but why use `id_info_id` rather than recording the type information with the name directly?

Reply:
talked in person

Reply:
We should not use id_info_id. This is not the purpose of id_info_id. See IdInfo.ml in deep-semgrep.
The main goal of id_info_id is to allow to modify back the generic AST while working on a view of the generic AST (SAST).

Reply:
I'm not sure I see the contradiction. Even if the purpose of `id_info_id` is to modify back the generic AST (because each identifier will contain its own `id_info_id`), I think these uses overlap -- it is necessary for us to produce a unique identifier that we can use to distinguish different instances of identifiers. One already exists, though. It's `id_info_id`. What would you propose instead?

Reply:
Furthermore , it's highly convenient. At the point of adding an identifier to the local scope in `Naming_SAST`, we add it only once we are already holding its `id_info_id`. I don't see too much of a point of generating a fresh ID, for instance, which will just exactly correspond to that `id_info_id`. I see `id_info_id` as a pointer which identifies certain `DefStmt`s, that we can use for debugging purposes and manual inspection also.

Reply:
An `id_info_id` identifies the variable occurrence not the variable.